pyecm.py

```diff
@@ -1,1483 +1,1569 @@
-The idea is to do many curves in parallel to take advantage of batch inversion algorithms. This gives asymptotically 7 modular multiplications per bit.
+We are using curves in Suyama's parametrization, but points are in affine coordinates,
+and the curve is in Wierstrass form.
+The idea is to do many curves in parallel to take advantage of batch inversion algorithms.
+This gives asymptotically 7 modular multiplications per bit.
+
+WARNING: pyecm is NOT a general-purpose number theory or elliptic curve library.
+Many of the functions have confusing calling syntax, and some will rather unforgivingly crash or return bad output
+if the input is not formatted exactly correctly.
 
-WARNING: pyecm is NOT a general-purpose number theory or elliptic curve library. Many of the functions have confusing calling syntax, and some will rather unforgivingly crash or return bad output if the input is not formatted exactly correctly. That said, there are a couple of functions that you CAN safely import into another program. These are: factors, isprime. However, be sure to read the documentation for each function that you use.
-'''
+That said, there are a couple of functions that you CAN safely import into another program.
+
+These are: factors, is_prime. However, be sure to read the documentation for each function that you use.
+"""
 
 import math
 import sys
 import random
 
-try:
-   import psyco
-   psyco.full()
-   PSYCO_EXISTS = True
-except ImportError:
-   PSYCO_EXISTS = False
-
-try: # Try to use gmpy
-   from gmpy2 import isqrt as sqrt
-   from gmpy2 import iroot as root
-   from gmpy2 import gcd, invert, mpz, next_prime
-   import gmpy2
-   GMPY_EXISTS = True
+try:  # Try to use gmpy
+    from gmpy2 import isqrt as sqrt
+    from gmpy2 import iroot as root
+    from gmpy2 import gcd, invert, mpz, next_prime
+    import gmpy2
+
+    GMPY_EXISTS = True
 except ImportError:
-   try:
-      from gmpy import gcd, invert, mpz, next_prime, sqrt, root
-      GMPY_EXISTS = True
-   except ImportError:
-      GMPY_EXISTS = False
+    GMPY_EXISTS = False
 
 if not GMPY_EXISTS:
-   PRIMES = (5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101, 103, 107, 109, 113, 127, 131, 137, 139, 149, 151, 167)
-   GMPY_EXISTS = False
-
-   def gcd(a, b):
-      '''Computes the Greatest Common Divisor of a and b using the standard quadratic time improvement to the Euclidean Algorithm.
+    PRIMES = (5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101, 103, 107,
+              109, 113, 127, 131, 137, 139, 149, 151, 167)
 
-Returns the GCD of a and b.'''
+    def gcd(number1, number2):
+        """
+        Computes the Greatest Common Divisor of a and b using the standard quadratic time
+        improvement to the Euclidean Algorithm. Returns the GCD of a and b.
+        """
 
-      if b == 0:
-         return a
-      elif a == 0:
-         return b
+        if number2 == 0:
```

Comment:
(1) As before, please no extraneous whitespace changes; here an unnecessary space is added. This applies to the rest of the file as well.

(2) I prefer `a` and `b` to `number1` and `number2`, as they are standard notation for GCD, and the new names convey no additional information.